src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.SymbolMapBuilder.cs

```diff
@@ -13,66 +13,74 @@
 
 namespace Microsoft.CodeAnalysis.ExtractMethod;
 
-internal abstract partial class MethodExtractor<TSelectionResult, TStatementSyntax, TExpressionSyntax>
+internal abstract partial class AbstractExtractMethodService<
+    TValidator,
+    TExtractor,
+    TSelectionResult,
+    TStatementSyntax,
+    TExpressionSyntax>
 {
-    protected abstract partial class Analyzer
+    internal abstract partial class MethodExtractor
```

Comment:
all these helper types became nested types.  they could all stop being generic by doing that.